networking_ccloud/ml2/agent/eos/switch.py

```diff
@@ -91,282 +172,342 @@ def get_switch_status(self):
+            for iface in ifaces or []:
+                # vlan stuff (native vlan, trunk vlans, translations)
+                data_vlan = {}
+
+                # native vlan
+                if iface.native_vlan:
+                    data_vlan['native-vlan'] = iface.native_vlan
+
+                # trunk vlans
+                if iface.trunk_vlans:
+                    data_vlan['interface-mode'] = 'TRUNK'
+                    data_vlan['trunk-vlans'] = iface.trunk_vlans
+
+                # vlan translations
+                if iface.vlan_translations:
+                    data_vlan['vlan-translation'] = {"ingress": [], "egress": []}
+                    for vtrans in iface.vlan_translations:
+                        data_vlan['vlan-translation']['ingress'].append(
+                            {"translation-key": vtrans.outside,
+                             "config": {"translation-key": vtrans.outside, "bridging-vlan": vtrans.inside}})
+                        data_vlan['vlan-translation']['egress'].append(
+                            {"translation-key": vtrans.inside,
+                             "config": {"translation-key": vtrans.inside, "bridging-vlan": vtrans.outside}})
+
+                # port-channel configuration
+                normal_ifaces = []
+                if iface.portchannel_id is not None:
+                    data = {
+                        'config': {
+                            'arista-intf-augments:mlag': iface.portchannel_id,
+                            'lag-type': 'LACP',
+                            'arista-intf-augments:fallback': 'individual',
+                        },
+                    }
+                    if data_vlan:
+                        data['switched-vlan'] = {'config': data_vlan}
+
+                    pc_cfg = (EOSGNMIClient.PATHS.IFACE_PC.format(iface=iface.name), data)
+                    config_req.get_list(operation).append(pc_cfg)
+                    normal_ifaces = iface.members or []
+                else:
+                    normal_ifaces = [iface.name]
+
+                for iface_name in normal_ifaces:
+                    data = {}
+                    if iface.portchannel_id:
+                        data['config'] = {'aggregate-id': f'Port-Channel{iface.portchannel_id}'}
+                    if data_vlan:
+                        data['switched-vlan'] = {'config': data_vlan}
+                    iface_cfg = (EOSGNMIClient.PATHS.IFACE_ETH.format(iface=iface_name), data)
+                    config_req.get_list(operation).append(iface_cfg)
+        else:
+            # delete everything (that is requested)
+            def calc_delete_range(all_ifaces, iface_name, iface_cfg):
+                if iface_name not in ifaces:
+                    return []
+                # FIXME: could this list get too big? (for sending the request of unpacked vlans)
+                return list(set(ifaces[iface_name].trunk_vlans) - set(iface_cfg.trunk_vlans))
```

Comment:
You're getting the parameter `all_ifaces` here, but are not using it and rely instead onto the outside-scopes function-parameter `ifaces` - is that intended (because it doesn't look like it, as you're passing in `all_ifaces_cfg` down below which you were explicitly retrieving before)?